vocode/streaming/synthesizer/deepgram_synthesizer.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+from typing import Optional
+
+from deepgram import DeepgramClient, DeepgramClientOptions, SpeakOptions
+
+from vocode.streaming.models.audio import AudioEncoding, SamplingRate
+from vocode.streaming.models.message import BaseMessage
+from vocode.streaming.models.synthesizer import DeepgramSynthesizerConfig
+from vocode.streaming.synthesizer.base_synthesizer import BaseSynthesizer, SynthesisResult
+from vocode.streaming.utils.create_task import asyncio_create_task_with_done_error_log
+from vocode.streaming.utils.mp3_helper import decode_mp3
+
+
+class DeepgramSynthesizer(BaseSynthesizer[DeepgramSynthesizerConfig]):
+    def __init__(
+        self,
+        synthesizer_config: DeepgramSynthesizerConfig,
+        api_key: Optional[str] = "",
+    ):
+        super().__init__(synthesizer_config)
+
+        # save config
+        self.synthesizer_config = synthesizer_config
+
+        self._config: SpeakOptions = SpeakOptions()
+        self._config.encoding = self.synthesizer_config.audio_encoding
+
+        if not self._config.model:
+            self._config.model = self.synthesizer_config.model
+        if not self._config.bit_rate:
+            self._config.bit_rate = self.synthesizer_config.bit_rate
+        if not self._config.container:
+            self._config.container = self.synthesizer_config.container
+        if not self._config.sample_rate:
+            self._config.sample_rate = self.synthesizer_config.sampling_rate
+
+        # deepgram client
+        config: DeepgramClientOptions = DeepgramClientOptions()
+        deepgram: DeepgramClient = DeepgramClient(api_key, config)
+        if not deepgram.api_key:
+            raise Exception(
+                "Please set DEEPGRAM_API_KEY environment variable or pass it as a parameter"
+            )
+        self._dgClient = deepgram.asyncspeak.v("1")
+
+        # off script options
+        self.experimental_streaming = synthesizer_config.experimental_streaming
+
+    async def create_speech_uncached(
+        self,
+        message: BaseMessage,
+        chunk_size: int,
+        is_first_text_chunk: bool = False,
+        is_sole_text_chunk: bool = False,
+    ) -> SynthesisResult:
+        # if self.experimental_streaming:
+        #     # TODO: streaming
+        #     return SynthesisResult(
+        #         self.experimental_mp3_streaming_output_generator(
+        #             response, chunk_size, create_speech_span
+        #         ),  # should be wav
+        #         lambda seconds: self.get_message_cutoff_from_voice_speed(
+        #             message, seconds, self.words_per_minute
+        #         ),
+        #     )
+        # else:
+        # TODO: indent when implementing streaming
+        try:
+            response = await self._dgClient.stream({"text": message.text}, options=self._config)
```

Comment:
i'd also love to get this working more efficiently by using streaming - seems like the DG client supports this natively.

mind following the pattern of the 11labs synthesizer here[0]? will make the experience so much better, but can also be done later, no worries

[0] https://github.com/vocodedev/vocode-core/blob/53b01dab0b59f71961ee83dbcaf3653a6935c2e3/vocode/streaming/synthesizer/eleven_labs_synthesizer.py#L125-L160

Reply:
I will see if we can end up supporting this better with the current implementation. Just an FYI, we have a streaming/WebSocket interface coming out soon which will likely obsolete this work/code.

Reply:
FYI, We are super close to that release (in days not weeks) which I will hold for and implement using that.

Reply:
got it. how much different is the SDK version with callbacks vs. our current websocket implementation? curious if we'll get performance benefits.

Reply:
this is mainly on the TTS side. will get a websocket implementation instead.

the threaded version is much faster than the async version. I also can make enhancements to this implementation to push the events to a queue on the classes in this project so it negates the time on the callback

Reply:
sorry, I meant on the STT side!